recipes/zmqpp/all/conanfile.py

```diff
@@ -0,0 +1,87 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class ZmqppConan(ConanFile):
+    name = "zmqpp"
+    homepage = "https://github.com/zeromq/zmqpp"
+    license = "MPL-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "This C++ binding for 0mq/zmq is a 'high-level' library that hides most of the c-style interface core 0mq provides."
+    topics = ("conan", "zmq", "0mq", "ZeroMQ", "message-queue", "asynchronous")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("libsodium/1.0.18")
+        self.requires("zeromq/4.3.3")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("zmqpp-%s" % (self.version), self._source_subfolder)
+
+    def _patch_sources(self):
+        cmakeFile = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        # zmqpp misses find for sodium
+        tools.replace_in_file(cmakeFile, "enable_testing()", "#enable_testing()\n"
+                                         "find_package(libsodium REQUIRED)")
+        if self.options.shared:
+            # zmqpp misses linking to sodium
+            tools.replace_in_file(cmakeFile, "set( LIB_TO_LINK_TO_EXAMPLES zmqpp )",
+                                             "set( LIB_TO_LINK_TO_EXAMPLES zmqpp )\n"
+                                             "target_link_libraries(zmqpp sodium)")
+        else:
+            # zmqpp uses different name for static library
+            tools.replace_in_file(cmakeFile, "generate_export_header(zmqpp)", "generate_export_header(zmqpp-static)")
+            tools.replace_in_file(cmakeFile, "${CMAKE_CURRENT_BINARY_DIR}/zmqpp_export.h",
+                                             "${CMAKE_CURRENT_BINARY_DIR}/zmqpp-static_export.h")
```

Comment:
I struggling to understand this patch 

https://github.com/zeromq/zmqpp/blob/85ae96020f2376c53d2176e04e88e8e51021b748/CMakeLists.txt#L149-L152

It looks like it's already taken care of 🤔

Reply:
I fetch sources from zmqpp tag 4.2.0, "stable" version. And the link above is for develop branch.

Reply:
You are absolutely correct.

Since the was fixed upstream, you should apply the changes in a separate patch (or else it will break with new versions)

Thanks for helping explain the issue to me 🤗

Reply:
https://github.com/zeromq/zmqpp/blob/f8ff127683dc555aa004c0e6e2b18d2354a375be/CMakeLists.txt#L148

Reply:
Ok, so I should create and keep this patch with the recipe and apply it before build. Am I correct?

Reply:
On mobile so I can't get a good example...

If you check `recipes/jwt-cpp` that is a good example, we make explicit `.patch` files and applied them during the build method

Reply:
Thank you for clarification!